### PR TITLE
#3 family members

### PR DESCRIPTION
**Pull request by ewaczajka**
## Changes:
- added new view - family members - where it's possible to create, edit or delete members in active family

## To do:
- saving selected family member in context

### CODE REVIEW

File: src/views/RoutesComponent.js [hunk 1]
```diff
@@ -8,34 +8,60 @@ import { SignUp } from 'views/SignUp/SignUp'
...
 
 export const RoutesComponent = () => {
 	const { activeFamily } = useContext(FamilyContext)
+	const { activeUser } = useContext(UserContext)
 
 	return (
 		<Routes>
 			<Route
 				path='/'
 				element={
```
Comment (nataliawalasik) [inline src/views/RoutesComponent.js:21]: I would go for something like this:
```
const RouteElement = ({ activeUser, activeFamily, children }) => {
  return activeUser ? (
    children
  ) : activeFamily ? (
    <Navigate replace to="/family" />
  ) : (
    <Navigate replace to="/login" />
  );
};
```

``` element={
          <RouteElement activeFamily={activeFamily} activeUser={activeUser}>
            <Dashboard />
          </RouteElement>
        }
```

Comment (nataliawalasik) [reply src/views/RoutesComponent.js:21]: And replace all the duplicated components in the file

File: src/components/molecules/UserEditor/UserEditor.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	SaveButton,
+} from './UserEditor.styles'
+
+export const UserEditor = ({
+	id,
+	name,
+	selectedColor,
+	deleteUser,
+	handleName,
+	handleColor,
```
Comment (nataliawalasik) [inline src/components/molecules/UserEditor/UserEditor.js:20]: handleChangeName, handleChangeColor,

File: src/views/FamilyMembers/FamilyMembers.js [hunk 3]
```diff
@@ -8,38 +8,153 @@ import {
...
+		}
+	}
+
+	const saveUser = async id => {
+		if (!currentUser.name) {
+			setError('Name is required!')
+		} else {
+			if (editing) {
+				setEditing(false)
+				const userDoc = doc(db, 'members', id)
```
Comment (nataliawalasik) [inline src/views/FamilyMembers/FamilyMembers.js:69]: I think that all the db-related logic could me moved to another component

File: src/components/organisms/Navbar/Navbar.js [hunk 4]
```diff
@@ -13,10 +13,10 @@ export const Navbar = () => {
 
 	const { activeFamily } = useContext(FamilyContext)
 	const { activeUser } = useContext(UserContext)
-	console.log(activeUser)
 
 	const signout = () => {
 		signOut(auth)
+		window.localStorage.removeItem('active_user_id')
```
[the remark this answers is not in the document]
Comment (ewaczajka) [reply src/components/organisms/Navbar/Navbar.js]: done

File: src/providers/ActiveUserProvider.js [hunk 5]
```diff
@@ -4,11 +4,22 @@ export const UserContext = createContext({
 	activeUser: '',
 })
 
+const KEY = 'active_user_id'
```
[the remark this answers is not in the document]
Comment (ewaczajka) [reply src/providers/ActiveUserProvider.js]: done

File: src/providers/ActiveUserProvider.js [hunk 6]
```diff
@@ -4,11 +4,22 @@ export const UserContext = createContext({
...
+
 export const ActiveUserProvider = ({ children }) => {
-	const [activeUser, setActiveUser] = useState('')
+	const activeUserId = window.localStorage.getItem(KEY) || ''
+	const [user, setUser] = useState(activeUserId)
+
+	const setUserId = userId => {
+		if (userId) {
+			setUser(userId)
+			window.localStorage.setItem(KEY, userId)
```
[the remark this answers is not in the document]
Comment (ewaczajka) [reply src/providers/ActiveUserProvider.js]: done - please check if I understood you correctly

File: src/views/FamilyMembers/FamilyMembers.js [hunk 7]
```diff
@@ -109,6 +109,7 @@ export const FamilyMembers = () => {
 	}
 
 	const selectUser = id => {
+		window.localStorage.removeItem('active_user_id')
```
[the remark this answers is not in the document]
Comment (ewaczajka) [reply src/views/FamilyMembers/FamilyMembers.js]: done

File: src/components/molecules/UserEditor/UserEditor.js (hunk 2 above)
Comment (ewaczajka) [reply src/components/molecules/UserEditor/UserEditor.js]: done

File: src/views/RoutesComponent.js (hunk 1 above)
Comment (ewaczajka) [reply src/views/RoutesComponent.js:30]: done

Comment (nataliawalasik) [reply src/views/RoutesComponent.js:30]: It has to be moved:
1. Outside of the component (eq. above if you want to keep it in the file)
or 
2. Into the separate file

I would go for second option ;)

File: src/providers/ActiveUserProvider.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (nataliawalasik) [reply src/providers/ActiveUserProvider.js]: yes! good job

**Review by nataliawalasik — APPROVED** [review]

**Pull request merged**